FEAT - SHA-256 checksumming

carlwilson opened the pull request:
- added `SHA-256` support to `Checksummer` class;
- added `SHA-256` instance to `ChecksumType` enum;
- updated comments to reflect calculation of SHA-256; and
- ignore `README` files when testing.

Closes #104 and includes work done in #386 thanks to @escowles

carlwilson committed:
FEAT - SHA-256 checksumming
- added `SHA-256` support to `Checksummer` class;
- added `SHA-256` instance to `ChecksumType` enum;
- updated comments to reflect calculation of SHA-256; and
- ignore `README` files when testing.

escowles reviewed (approved):
👍 This looks good to me, thanks for resurrecting this @carlwilson !

tledoux reviewed (approved):
Looks good to me too.

carlwilson commented:
It still requires me to come up with a test patch to add or ignore the new checksums on travis.

carlwilson committed:
Merge branch 'integration' into feat/sha-256

carlwilson committed:
FIX - Streamline tests
- fixed shellcheck warnings;
- using `/var/tmp` for executablility; and
- tightened test scripts.
